src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -755,19 +757,68 @@ object TezosDatabaseOperations extends LazyLogging {
   def writeProcessedEventsLevels(eventType: String, levels: List[BigDecimal]): DBIO[Option[Int]] =
     Tables.ProcessedChainEvents ++= levels.map(Tables.ProcessedChainEventsRow(_, eventType))
 
-  /** Inserts registered tokens to the table if empty
-    *
+  /** Inserts to the table if table is empty
+    * @param table slick TableQuery[_] to which we want to insert
     * @param rows the type of event to record
     * @return the number of entries saved added to the
     */
-  def writeRegisteredTokensRowsIfEmpty(
-      rows: List[Tables.RegisteredTokensRow]
+  def insertIfEmptyTable[A <: AbstractTable[_]](
+      table: TableQuery[A],
+      rows: List[A#TableElementType]
   )(implicit ec: ExecutionContext): DBIO[Option[Int]] =
-    Tables.RegisteredTokens.exists.result.flatMap {
+    table.exists.result.flatMap {
       case true => DBIO.successful(Some(0))
-      case false => Tables.RegisteredTokens ++= rows
+      case false => table ++= rows
     }
 
+  import shapeless._
+  import shapeless.ops.hlist._
+
+  /** Trims if passed a String value, otherwise returns the value unchanged */
+  object Trimmer extends Poly1 {
+    implicit val stringTrim = at[String] { _.trim }
+    implicit def noop[T] = at[T] { identity }
+  }
+
+  /** Uses a Generic to transform the instance into an HList, maps over it and convert it back into the case class */
```

Comment:
I know it wasn't like this before, but I think we might want to move in the direction of

 * specifically define what the function does: "maps over ..." seems quite general
 * add doc line for types used in each param (even implicit)
 * add a short wiki-entry to descrive how Generics work and how we apply them here, for people with no experience in scala and shapeless

Reply:
What about this @piotrkosecki 
Do you plan anything?

Reply:
That's the comment from stackoverflow. I used the code from the so I left the original comment. I'm not an expert in shapeless 😛